addons/l10n_dk/models/account_account.py

```diff
@@ -0,0 +1,19 @@
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAccount(models.Model):
+    _inherit = ['account.account']
+    
+    @api.ondelete(at_uninstall=False)
+    def _unlink_bank_cash_accounts(self):
+        if self.env.company.country_code != 'DK':
+            return
+
+        nb_bank_cash_account = self.search_count([
+            ('company_id', '=', self.env.company.id),
+            ('account_type', '=', 'asset_cash'),
+        ])
+        nb_to_delete = sum(1 for account in self if account.account_type == 'asset_cash')
```

Comment:
Strange way of doing that but it works ahah 
GG @SaddemAmine 

For the record, we can use filtered as well.

```suggestion
        nb_to_delete = len(self.filtered(lambda account: account.account_type == 'asset_cash'))
```